### ISSUE TITLE
Launching Windows application with command-line args with Appium

### ISSUE THREAD

**Issue by samuelfreiberg** [microsoft/WinAppDriver#1140]
I'm attempting to use Appium to test one of my Windows applications. Currently, I need to provide the path to my Win32 application, **but**, in order to run my application in the correct state, I need to pass command-line arguments to my Win32 application. I see there's a "processArguments" capability for iOS that allows you to pass arguments to the application. Is there something similar for Windows? How can I achieve passing command-line args to my Win32 app? Any help would be much appreciated!

**Comment by Tree55Topz** [comment]
I am not sure what this has to do with WinAppDriver, but I am able to achieve this on my app using the C# Process class.

Process.Start(exePath, " --processStart \"launch.exe\" --process-start-args \"enableautomation\"");

where exePath is the path to the executable file, and enableautomation is an arg developers added to populate automation IDs on the host

**Comment by samuelfreiberg** [comment]
I should have been more specific - Appium uses WinAppDriver to open my application. Thus, I believe this is a WinAppDriver inquiry. 

I found that WinAppDriver allows an "appArguments" capability, but I can't seem to make it work. I added --process-start-args but that didn't seem to fix it. Thank you for the response though. Any other thoughts?

**Comment by Tree55Topz** [comment]
Ahh got you, yes try something like this...

```
AppiumOptions appCapabilities = new AppiumOptions();
appCapabilities.AddAdditionalCapability("appArguments", "enableautomation");
session = new WindowsDriver<WindowsElement>(new Uri(WindowsApplicationDriverUrl), appCapabilities, TimeSpan.FromMinutes(2));
```

**Comment by samuelfreiberg** [comment]
Are "appArguments" the same as command-line arguments? I can't seem to find much information on them

**Comment by Tree55Topz** [comment]
Yes I believe so. In my examples above, the second one is how I was able to include the command line argument of "enableautomation" on my application. 

Did you try this?

**Comment by Tree55Topz** [comment]
I believe they are the same thing... This doc has some additional examples such as using ones specified in a file

https://github.com/microsoft/WinAppDriver/blob/master/Docs/AuthoringTestScripts.md
